src/autorun.ts

```diff
@@ -94,10 +142,11 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
         deps.clear();
     });
 
-    function createTracker(track: boolean) {
+    function createTracker(track: boolean, trackType: TrackType) {
         return function $<O>(o: Observable<O>): O {
             if (deps.has(o)) {
                 const v = deps.get(o)!;
+                v.used = true;
                 if (track && !v.track) {
```

Comment:
**minor:** I think, we'll also need to raise the `trackType` to max of two. As we raise the `_` vs `$` on second visit.
In cases similar to `run(() => $.weak(a) + $(a))` (pseudocode)

Reply:
Good one! Hmm that would involve some more complex test scenarios I think... Lets see.